packages/terra-paginator/src/Paginator.jsx

```diff
@@ -120,22 +119,18 @@ class Paginator extends React.Component {
     return pageButtons;
   }
 
-  hasNavContext() {
-    return this.props.totalCount && this.props.itemCountPerPage;
-  }
-
   defaultPaginator() {
     const theme = this.context;
     const { intl } = this.props;
-    const totalPages = calculatePages(this.props.totalCount, this.props.itemCountPerPage);
+    const totalPages = (this.props.totalCount) ? calculatePages(this.props.totalCount, this.props.itemCountPerPage) : 0;
     const { selectedPage } = this.state;
     const previousPageIndex = selectedPage === 1 ? 1 : selectedPage - 1;
     const nextPageIndex = selectedPage === totalPages ? totalPages : selectedPage + 1;
 
     const fullView = (
-      <div className={cx('paginator', !this.hasNavContext() && 'pageless', theme.className)}>
+      <div className={cx('paginator', !this.props.totalCount && 'pageless', theme.className)}>
```

Comment:
Would `!this.hasNavContext()` be valid when totalCount is provided?

Reply:
yes but hasNavContext() does check for `itemCountPerPage` so it will be valid only when `totalCount` and `itemCountPerPage` are provided